mergify_engine/queue/queue_freeze.py

```diff
@@ -0,0 +1,143 @@
+# -*- encoding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import dataclasses
+import datetime
+import typing
+
+import daiquiri
+import msgpack
+
+from mergify_engine import date
+from mergify_engine import utils
+from mergify_engine.context import Repository
+from mergify_engine.web.api import queues
+
+
+LOG = daiquiri.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class QueueFreeze:
+
+    repository: Repository
+
+    # Stored in redis
+    name: str = dataclasses.field(metadata={"description": "Queue name"})
+    reason: str = dataclasses.field(metadata={"description": "Freeze reason"})
+    user_app_name: str = dataclasses.field(
+        metadata={"description": "Application responsible for freeze"}
+    )
+    freeze_date: datetime.datetime = dataclasses.field(default_factory=date.utcnow)
+
+    class Serialized(typing.TypedDict):
+        name: str
+        reason: str
+        user_app_name: str
+        freeze_date: datetime.datetime
+
+    @classmethod
+    async def get_all(cls, repository: Repository) -> typing.List[queues.QueueFreeze]:
+
+        queue_freeze = cls(
+            repository=repository,
+            name="",
+            reason="",
+            user_app_name="",
+            freeze_date=datetime.datetime.utcnow(),
+        )
+
+        qf_list = []
+        with utils.yaaredis_for_queue() as redis_queue:
+            async for key, qf_raw in redis_queue.hscan_iter(
+                name=queue_freeze._get_redis_hash(),
+                match=queue_freeze._get_redis_key_match(),
+            ):
+                qf = msgpack.unpackb(qf_raw, timestamp=3)
+                qf_list.append(
+                    queues.QueueFreeze(
+                        name=qf["name"],
+                        reason=qf["reason"],
+                        user_app_name=qf["user_app_name"],
+                        freeze_date=qf["freeze_date"],
+                    )
+                )
+
+        return qf_list
```

Comment:
I would rather `yield` those rather than build the list; if the caller does not need the full list, we'll save time and memory.